### PR TITLE
Test interact

### PR DESCRIPTION
**Pull request by huard**
## Overview

Added tests for interact. Some refactoring. 

Changes:

* Added support for ComplexData in widgets
* Renamed convert_input and convert_output
* Add test for interact

### CODE REVIEW

File: birdy/client/utils.py [hunk 1]
```diff
@@ -111,6 +112,8 @@ def convert_output_value(value, data_type):
         value = dateutil.parser.parse(value).time()
     elif "date" in data_type:
         value = dateutil.parser.parse(value).date()
+    elif "angle" in data_type:
```
Comment (cehbrecht) [inline birdy/client/utils.py]: Why is it `angle`? Shouldn't it be `float`?
```
//www.w3.org/TR/xmlschema-2/#float
```

Comment (huard) [reply birdy/client/utils.py]: https://pywps.readthedocs.io/en/master/_modules/pywps/inout/literaltypes.html

Comment (cehbrecht) [reply birdy/client/utils.py]: ok.

**Comment by cehbrecht** [comment]
done.

**Pull request merged**